src/containers/rbtree/rbtree.h

```diff
@@ -0,0 +1,794 @@
+#include <algorithm>
+#include <iomanip>
+#include <iostream>
+template <typename K, typename T>
+class rbtree {
+ public:
+  using key_type = K;
+  using mapped_type = T;
+  using value_type = std::pair<const key_type, mapped_type>;
+  using reference = value_type&;
+  using const_reference = const value_type&;
+  using size_type = size_t;
+  using pointer = value_type*;
+  struct rbtree_node {
+    rbtree_node(const value_type& value)
+        : value_(value),
+          right_(nullptr),
+          left_(nullptr),
+          parent_(nullptr),
+          mColor('R') {}
+    value_type value_;
+    rbtree_node* right_;
+    rbtree_node* left_;
+    rbtree_node* parent_;
+    char mColor;
+  };
+
+  using node = rbtree_node;
+
+ protected:
+  struct rbtree_iterator {
+   public:
+    rbtree_iterator() noexcept;
+    rbtree_iterator(node* node) noexcept;
+    rbtree_iterator(const rbtree_iterator& other) : current_(other.current_) {};
+    reference operator*() const;
+    pointer operator->() const;
+    rbtree_iterator& operator++();
+    rbtree_iterator operator++(int);
+    rbtree_iterator& operator--();
+    rbtree_iterator operator--(int);
+    bool operator==(const rbtree_iterator& lhs);
+    rbtree_iterator& operator=(const rbtree_iterator& other) {
+      if (this != &other) {
+        current_ = other.current_;
+      }
+      return *this;
+    }
+    bool operator!=(const rbtree_iterator& lhs);
+
+   private:
+    friend class rbtree;
+    node* current_;
```

Comment:
Мне кажется, не совсем удачное имя. Что значит "текущий"? Что будет значить `текущий = другой.текущий`, как в 45 строке? Меня это путает, честно говоря. Если это "нода", то и назвать её следует "нодой".

```suggestion
    node* node_;
```